src/main/java/com/cloudbees/jenkins/github/AccessTokenCredential.java

```diff
@@ -0,0 +1,109 @@
+package com.cloudbees.jenkins.github;
+
+import com.cloudbees.plugins.credentials.CredentialsDescriptor;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.AutoCompletionCandidates;
+import hudson.util.FormValidation;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import org.kohsuke.github.GHAuthorization;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public class AccessTokenCredential extends BaseStandardCredentials implements StandardUsernameCredentials {
+
+    private final String githubServer;
+
+    private final String username;
+
+    private final Secret token;
```

Comment:
Does credentials plugin has some base class for tokens, so github-plugin would extend it only with it's unique internals?
What is really needed for storing tokens? Isn't enough for plugin to store server url separately and add token (but with credentials plugin) to configuration?